pages/products/index.js

```diff
@@ -0,0 +1,119 @@
+import { SVGIcon } from "@/components/SVGIcon";
+import { useAtom } from "jotai";
+import { RESET } from "jotai/utils";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+
+import { initialProducts } from "../product/[code]";
+
+const StyledAllCards = styled.div`
+  margin-left: 0.4rem;
+  width: 97vw;
+  margin-top: 4rem;
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  gap: 0.5rem;
+`;
+const StyledImage = styled(Image)`
+  width: 135px;
+  height: 135px;
+  object-fit: contain;
+`;
+const StyledProductCard = styled.section`
+  witdh: 50%;
+  //   margin-left: 3%;
+  //   width: 80%;
+  background-color: white;
+  //   margin-left: 1rem;
+  //   margin-right: -0.7rem;
+  border-radius: 0.4rem;
+  //   box-shadow: 1px 4px 10px 1px rgb(127, 133, 136);
+  //   height: 100%;
+`;
+const StyledProductName = styled.h2`
+  border-radius: 0.4rem 0.4rem 0 0;
+  background: #ffcc80;
+  padding: 0.5rem;
+  font-weight: bold;
+  font-size: 1.2rem;
+  text-align: center;
+`;
+
+const StyledDeleteButton = styled.button`
+  background: none;
+  border: none;
+  padding: 0;
+  font-size: inherit;
+  cursor: pointer;
+  color: inherit;
+`;
+
+const StyledDetailsButton = styled.button`
+  background: none;
+  border: none;
+  padding: 0;
+  font-size: inherit;
+  cursor: pointer;
+  color: inherit;
+`;
+
+const StyledButtonPosition = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const StyledInsideCard = styled.div``;
+
+export default function Products() {
+  const [products, setProducts] = useAtom(initialProducts);
+  const router = useRouter();
+
+  return (
+    <>
+      <StyledAllCards>
+        {products.map((product) => {
+          return (
+            <StyledProductCard key={product.code}>
+              <StyledProductName>{product.name}</StyledProductName>
+              <StyledButtonPosition>
+                <StyledDetailsButton
+                  onClick={() => {
+                    router.push(`/product/${product.code}`);
+                  }}
+                >
+                  <SVGIcon variant="details" width="26px" />
+                </StyledDetailsButton>
```

Comment:
This should be a link component. Please use the next/link component here instead with the `href` prop.

Reply:
done

> Hey Sascha, I added some suggestions for your code, have a look. Generally already good stuff!

thaanks :)